main.go

```diff
@@ -50,19 +51,20 @@ func main() {
 
 	err = validation.Validate(opts)
 	if err != nil {
-		logger.Printf("%s", err)
-		os.Exit(1)
+		logger.Fatalf("%s", err)
 	}
 
 	validator := NewValidator(opts.EmailDomains, opts.AuthenticatedEmailsFile)
 	oauthproxy, err := NewOAuthProxy(opts, validator)
 	if err != nil {
-		logger.Errorf("ERROR: Failed to initialise OAuth2 Proxy: %v", err)
-		os.Exit(1)
+		logger.Fatalf("ERROR: Failed to initialise OAuth2 Proxy: %v", err)
 	}
 
 	rand.Seed(time.Now().UnixNano())
 
+	http.HandleFunc("/metrics", middleware.MetricsServe)
+	go http.ListenAndServe(opts.MetricsPort, nil)
```

Comment:
Is there a way to have this stop gracefully as we do in the main server?

Reply:
Should we also have this be opt-in and disabled from this metrics server port being turned on by default?

For users who don't have prometheus in their environment, it would be best to not open up extra unneeded ports from a security perspective.

Reply:
@JoelSpeed I can look into making this more graceful.

@NickMeves We could if that is what is desired. With this being on a different port, it shouldn't be exposed if the user has correctly set up their firewall/docker ports. What should be the default mode, metrics on by default or off?

Reply:
> @JoelSpeed I can look into making this more graceful.

This could be left as a TODO if it's a lot of work at this point, I'll let you make this call

> What should be the default mode, metrics on by default or off?

Off by default IMO, perhaps a special value of `0` makes it off and is the default value?

Reply:
":0" technically means bind to a random free port in Windows and Linux. I could change the default to an empty string and only enable the server when we are provided an address. Also might be worth changing the flag to address instead of port as well.

Reply:
I lean toward a dedicated flag on whether metrics is enabled or disabled. Then a separate flag specifying the port to listen on if enabled.

Reply:
Perhaps we should change this to `MetricsBindAddress` and accept a host and port, then have empty string as off, eg I could set `127.0.0.1:8080` for only binding to localhost, is that relatively easy to achieve?

Reply:
I didn't see this comment but I have essentially implemented this I can rename the function.